Introduce `gh agent-task`

BagToad opened the pull request:
Introduce `gh agent-task` with the following aliases:

- `gh agent-tasks`
- `gh agent`
- `gh agents`

This establishes the root command for future `gh agent-task` sub-commands.

This also implements a persistent OAuth token requirement that will run for all future sub-commands. 

> [!IMPORTANT]
> NOT merging into `trunk`

BagToad committed:
Add agent-task command with OAuth token validation

Introduces a new `agent-task` command under pkg/cmd/agent with strict OAuth (device flow) token validation. Includes comprehensive tests for token source and host validation, and registers the command in the root command set.

babakks commented on pkg/cmd/agent-task/agent_task_test.go:
```diff
@@ -0,0 +1,172 @@
+	require.Error(t, err)
+}
+
+func TestAliasAreSet(t *testing.T) {
```
I think this test is not needed.

BagToad replied:
Tell me more 😜 I want to ensure these aliases are not messed with. I wrote this test because I want these aliases to always be reliable, and users to feel comfortable automating their workflows with them.

babakks replied:
Yeah, I know what you say, but I don't think we'll ever remove any of those unknowingly so that the test stops us. If you feel strong about it, I'm okay to keep it though.

BagToad replied:
I don't feel strongly necessarily; However, I do feel strongly that tests like this don't cost us anything. It's simple, easy to read, clear what it's doing, negligible runtime, and the only way it would break is if we actually did break the alias (I hope 🤞 ). So, to me it's just free resiliency and makes absolutely clear that these aliases aren't an after thought, but instead are a critical behavior feature.

babakks commented on pkg/cmd/agent-task/agent_task.go:
```diff
@@ -0,0 +1,62 @@
+		// This is required to run this root command. We want to
+		// run it to test PersistentPreRunE behavior.
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
```
What happens if I run `gh agent-task`? I mean, we want to show the usage/docs in this case, right?

BagToad replied:
Yes! We should do that as a TODO ✅

BagToad replied:
Actually, fixed in 2128a297b37550b6c98e78f03a6ae1452799a2af

babakks commented on pkg/cmd/agent-task/agent_task_test.go:
```diff
@@ -0,0 +1,172 @@
+	cmd := NewCmdAgentTask(f)
+	err := cmd.Execute()
+	require.NoError(t, err)
+	require.Equal(t, "", stdout.String())
```
Can be `require.Empty`.

BagToad replied:
Good call!

BagToad replied:
Fixed in 1bc2710c883b27093115dba207febd78817a92aa

babakks commented on pkg/cmd/agent-task/agent_task_test.go:
```diff
@@ -0,0 +1,172 @@
+	return c
+}
+
+func TestOAuthTokenAccepted(t *testing.T) {
```
Why not a table test? I mean, to avoid dups.

BagToad replied:
I find table tests a bit hard to read personally. What do you think?

babakks replied:
I think in this case a TT helps.

BagToad replied:
Refactored in b939188e6ddc240d29420e95f14c1e109cf91479

babakks reviewed (approved):
Thanks for the PR! :pray: I think we should simplify the tests, but I'm also okay with a follow up. So, approving.

BagToad committed:
Show help on agent-task command execution

Changed the RunE function of the agent-task command to display help output instead of returning nil, improving user guidance when the command is run without arguments.

BagToad committed:
Refactor test to use require.Empty assertion

Replaces require.Equal with require.Empty in TestOAuthTokenAccepted for improved clarity when checking for empty output.

BagToad committed:
Refactor agent task tests into table-driven format

Consolidates multiple individual test functions into a single table-driven test, improving maintainability and readability. This change makes it easier to add new test cases and ensures consistent test structure for agent task command authentication scenarios.
